fearless/Common/DataProvider/SubstrateLocalSubscriptionFactory.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import RobinHood
+
+class SubstrateLocalSubscriptionFactory {
+    private var providers: [String: WeakWrapper] = [:]
+
+    let chainRegistry: ChainRegistryProtocol
+    let storageFacade: StorageFacadeProtocol
+    let operationManager: OperationManagerProtocol
+    let logger: LoggerProtocol
+    let stremableProviderFactory: SubstrateDataProviderFactoryProtocol
+
+    init(
+        chainRegistry: ChainRegistryProtocol,
+        storageFacade: StorageFacadeProtocol,
+        operationManager: OperationManagerProtocol,
+        logger: LoggerProtocol
+    ) {
+        self.chainRegistry = chainRegistry
+        self.storageFacade = storageFacade
+        self.operationManager = operationManager
+        self.logger = logger
+        stremableProviderFactory = SubstrateDataProviderFactory(
+            facade: storageFacade,
+            operationManager: operationManager,
+            logger: logger
+        )
+    }
+
+    func clearIfNeeded() {
+        providers = providers.filter { $0.value.target != nil }
+    }
```

Comment:
Should it be private?

Reply:
By default it is internal. And can't be private due to usage from subsclass